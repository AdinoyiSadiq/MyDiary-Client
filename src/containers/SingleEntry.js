import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getSingleEntry } from '../actions/entryActions';

class SingleEntry extends Component {
  componentWillMount() {
    const { getSingleEntry, match: { params: { id } } } = this.props;
    getSingleEntry(id);
  }

  renderSingleEntry() {
    const { entry } = this.props;
    if (entry) {
      const date = new Date(parseInt(entry.created));
      return (
        <article className="entry">
          <h3 className="title">{entry.title}</h3>
          <p className="date">{date.toString().substring(0, 15)}</p>
          <p className="content">{entry.content}</p>
          <div className="actions">
            <div title="Edit">
              <Link to={`/main/updateEntry/${entry.id}`}>
                <i className="far fa-edit"></i>
              </Link>
            </div>
					  <div title="Delete"><i className="far fa-trash-alt"></i></div>
          </div>
        </article>
      );
    }
    return <div>Loading...</div>
  }

  render() {
    return (
      <main>
        {this.renderSingleEntry()}
      </main>
    );
  }
};

function mapStateToProps(state) {
  return { entry: state.singleEntry.entry };
}

export default connect(mapStateToProps, { getSingleEntry })(SingleEntry);