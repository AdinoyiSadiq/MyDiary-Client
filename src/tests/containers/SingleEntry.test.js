import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import SingleEntry from '../../containers/SingleEntry';
import Root from '../../root';

let wrapped;
let loading;

const match = { 
  params: { id: 2 } 
}

const mockState = { 
  singleEntry: {
    entry: {}
  },
};

beforeEach(() => {
  wrapped = mount(
    <Root initialState={mockState}>
      <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
        <SingleEntry match={match} />
      </MemoryRouter>
    </Root>,
  );

  loading = mount(
    <Root>
      <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
        <SingleEntry match={match} />
      </MemoryRouter>
    </Root>,
  );
});

describe('Single Entry UI', () => {
  describe('render features', () => {
    test('container should render as expected when entry has been retrieved', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });

    test('container should render as expected when loading', () => {
      const tree = toJson(loading);
      expect(tree).toMatchSnapshot();
    });
  });
});  