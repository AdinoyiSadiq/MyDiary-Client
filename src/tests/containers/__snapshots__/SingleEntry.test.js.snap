// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Entry UI render features container should render as expected when entry has been retrieved 1`] = `
<_default
  initialState={
    Object {
      "singleEntry": Object {
        "entry": Object {},
      },
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          Object {
            "key": "testKey",
          },
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/",
                "search": "",
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(SingleEntry)
          match={
            Object {
              "params": Object {
                "id": 2,
              },
            }
          }
        >
          <SingleEntry
            deleteEntry={[Function]}
            entry={Object {}}
            getSingleEntry={[Function]}
            match={
              Object {
                "params": Object {
                  "id": 2,
                },
              }
            }
          >
            <main>
              <DeleteEntryModal
                deleteEntry={[Function]}
                display={false}
                hideDeleteModal={[Function]}
              >
                <section
                  className="authContent deleteModal null"
                >
                  <div
                    className="title"
                  >
                    Do you want to delete this entry?
                  </div>
                  <hr />
                  <div
                    className="clearfix"
                  >
                    <button
                      id="deleteButton"
                      onClick={[Function]}
                    >
                      Yes
                    </button>
                    <button
                      id="cancelButton"
                      onClick={[Function]}
                    >
                      No
                    </button>
                  </div>
                </section>
              </DeleteEntryModal>
              <article
                className="entry"
              >
                <h3
                  className="title"
                />
                <p
                  className="date"
                >
                  Invalid Date
                </p>
                <p
                  className="content"
                />
                <div
                  className="actions"
                >
                  <div
                    title="Edit"
                  >
                    <Link
                      replace={false}
                      to="/main/updateEntry/undefined"
                    >
                      <a
                        href="/main/updateEntry/undefined"
                        onClick={[Function]}
                      >
                        <i
                          className="far fa-edit"
                        />
                      </a>
                    </Link>
                  </div>
                  <div
                    onClick={[Function]}
                    title="Delete"
                  >
                    <i
                      className="far fa-trash-alt"
                    />
                  </div>
                </div>
              </article>
            </main>
          </SingleEntry>
        </Connect(SingleEntry)>
      </Router>
    </MemoryRouter>
  </Provider>
</_default>
`;

exports[`Single Entry UI render features container should render as expected when loading 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          Object {
            "key": "testKey",
          },
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/",
                "search": "",
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(SingleEntry)
          match={
            Object {
              "params": Object {
                "id": 2,
              },
            }
          }
        >
          <SingleEntry
            deleteEntry={[Function]}
            getSingleEntry={[Function]}
            match={
              Object {
                "params": Object {
                  "id": 2,
                },
              }
            }
          >
            <main>
              <DeleteEntryModal
                deleteEntry={[Function]}
                display={false}
                hideDeleteModal={[Function]}
              >
                <section
                  className="authContent deleteModal null"
                >
                  <div
                    className="title"
                  >
                    Do you want to delete this entry?
                  </div>
                  <hr />
                  <div
                    className="clearfix"
                  >
                    <button
                      id="deleteButton"
                      onClick={[Function]}
                    >
                      Yes
                    </button>
                    <button
                      id="cancelButton"
                      onClick={[Function]}
                    >
                      No
                    </button>
                  </div>
                </section>
              </DeleteEntryModal>
              <div>
                Loading...
              </div>
            </main>
          </SingleEntry>
        </Connect(SingleEntry)>
      </Router>
    </MemoryRouter>
  </Provider>
</_default>
`;
